test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs

```diff
@@ -68,6 +83,33 @@ public void ParseRouteParameter_ConstraintWithArgumentsAndOptional_ParsedCorrect
 
             Assert.Single(templatePart.InlineConstraints);
             Assert.Single(templatePart.InlineConstraints, c => c.Constraint == @"test(\d+)");
+
+            templatePart = ParseParameter(@"param:test(\d+)=abc?");
```

Comment:
**never** do this kind of thing. Convert it to data-driven test (`[Theory]`).

Also the second kind of case you added here is a different scenario right? One of these has a default value and one doesn't.